lib/modulesFromList.nix

```diff
@@ -43,14 +35,40 @@ let
 
       genAttrs' = func: values: builtins.listToAttrs (map func values);
 
+      hasFileAttr = o: builtins.hasAttr "_file" o;
+      peak = f: f (builtins.functionArgs f);
+
     in
 
     genAttrs'
-      (path: {
-        name = removeSuffix ".toml" (removeSuffix ".nix" (baseNameOf path));
-        value = _import path;
-      })
-      paths;
+      (arg:
+
+        # a regular path to be imported
+        if builtins.isPath arg then
+          {
+            name = removeSuffix ".nix" (baseNameOf arg);
+            value = import arg;
```

Comment:
I am not a big fan of the removal of `_import`. Was hoping to add recImportAttrs function here as well :/

Reply:
I consider this function to be more of an "exporter" if that makes sense. I think a recImport is an _importer_ and should be separate. We did an implementation for that in digga with rakeLeaves, maybe we can extract that into fup im just not sure where it would be useful for end users outside the digga context.